src/test/java/org/broadinstitute/hellbender/tools/walkers/coverage/DepthOfCoverageIntegrationTest.java

```diff
@@ -0,0 +1,259 @@
+        List<File> expectedFiles = Arrays.stream(Objects.requireNonNull(getExpectedDataDir().listFiles())).filter(f -> f.getName().contains(expectedBaseName)).sorted().collect(Collectors.toList());
+        List<File> actualFiles = Arrays.stream(actual).sorted().collect(Collectors.toList());
+
+        // First check that all the expected files correspond to actual files
+        List<String> expectedFilenames = expectedFiles.stream().map(f -> getDoCExtensionFromFile(f, expectedBaseName)).sorted().collect(Collectors.toList());
+        List<String> actualFilenames = actualFiles.stream().map(f -> getDoCExtensionFromFile(f, actualFileBaseName)).sorted().collect(Collectors.toList());
+        Assert.assertEquals(actualFilenames, expectedFilenames);
+
+        // Now assert that the outputs exactly match with the expected outputs
+        for (int i = 0; i < actual.length; i++) {
+            // Need this line for comparison to gatk3 in which the interval_summary file has non-deterministic header/sample line ordering
+            compareCSVByRowHeaderOrderAgnostic(actualFiles.get(i), expectedFiles.get(i), ',', actualFileBaseName);
+        }
+    }
+
+    // Helper class that allows us to get around mismatches sample line ordering that may mismatch in results files
+    private void compareCSVByRowHeaderOrderAgnostic( final File actualFile, final File expectedFile, char separator, String actualFileBaseName) {
+        try (final CSVReader actualParser = new CSVReader(new FileReader(actualFile), separator);
+             final CSVReader expectedParser = new CSVReader(new FileReader(expectedFile), separator);
+        ) {
+            List<String> actualHeader = Arrays.asList(actualParser.readNext());
+            List<String> expectedHeader = Arrays.asList(expectedParser.readNext());
+            // Deal with archaic standard of delimiting header with a different character from the body
+            if (expectedHeader.size() == 1) {
+                expectedHeader = new ArrayList<>(Arrays.asList(expectedHeader.get(0).split("[\t,]")));
+            }
+            //Assert headers match paying no heed to ordering
+            Assert.assertEquals(actualHeader.stream().sorted().iterator(), expectedHeader.stream().sorted().iterator());
+
+
+            Iterator<String[]> actualLines = StreamSupport.stream(actualParser.spliterator(),false).sorted(Comparator.comparing(l -> l[0]+","+l[1])).iterator();
+            Iterator<String[]> expectedLines = StreamSupport.stream(expectedParser.spliterator(),false).sorted(Comparator.comparing(l -> l[0]+","+l[1])).iterator();
+            // Special case to handle scrambled output header for the base coverage in gatk3
+            if (getDoCExtensionFromFile(actualFile, actualFileBaseName).equals("")) {
+                // Just assert all the correct counts are present for each line because the order is wrong in the expected files
```

Comment:
Doesn't the sort above compensate for ordering differences? I don't understand this comment.

Reply:
Yes, gatk3 test files included bugs that I fixed in terms of column ordering.